edgelet/hsm-sys/azure-iot-hsm-c/src/edge_pki_openssl.c

```diff
@@ -1881,3 +1882,38 @@ int verify_certificate
 
     return result;
 }
+
+int generate_rand_buffer(unsigned char *buffer, size_t num_bytes)
+{
+    int result;
+
+    initialize_openssl();
+
+    if ((buffer == NULL) || (num_bytes == 0) || (num_bytes > INT_MAX))

```

Comment:
I suppose it doesn't matter since `INT_MAX` is plenty large already, but you *could* support `num_bytes > INT_MAX` buffers by generating `INT_MAX` chunks separately. I say this because the function *does* take `size_t` and it's not obvious to callers that there's a soft `INT_MAX` limit.

Reply:
Sure something we can consider.